app/src/types.ts

```diff
@@ -0,0 +1,81 @@
+import { Dispatch, SetStateAction } from 'react'
+import { UrlObject } from 'url'
+
+/*
+ * Types relating to persistent user data
+ */
+export type Clinic = {
+  id: number
+  agency: string
+  agencyAddress: string
+  agencyTelephone: string
+  clinic: string
+  clinicAddress: string
+  clinicTelephone: string
+  county: string
+  zip: string
+}
+
+export type ChooseClinicData = {
+  clinic: Clinic | undefined
+  zipCode: string
+}
+
+export type ContactData = {
+  firstName: string
+  lastName: string
+  phone: string
+  comments: string
+}
+
+export type EligibilityData = {
+  residential: string
+  categorical: string[]
+  previouslyEnrolled: string
+  adjunctive: string[]
+}
+
+export type IncomeData = {
+  householdSize: string
+}
+
+export type SessionData = {
+  chooseClinic: ChooseClinicData
+  contact: ContactData
+  eligibility: EligibilityData
+  income: IncomeData
+  submitted: boolean
```

Comment:
Notes for reviwers: This is not user-input data, but tracks if the user has submitted the form and prevents a duplicate entries from being created if a user clicks the browser back button and clicks submit again.